lib/puppet/functions/dns_soa.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Retrieves DNS TXT records and returns it as an array. Each record in the
+# array will be a array containing the strings of the TXT record.
+#
+# An optional lambda can be given to return a default value in case the
+# lookup fails. The lambda will only be called if the lookup failed.
+Puppet::Functions.create_function(:dns_soa) do
+  dispatch :dns_soa do
+    param 'String', :record
+  end
+
+  dispatch :dns_soa_with_default do
+    param 'String', :record
+    block_param
+  end
+
+  def dns_soa(record)
+    Puppet.deprecation_warning('dns_soa', 'This method is deprecated please use the namspaced version dnsquery::soa')
```

Comment:
This method does not exist yet, so I think we can just remove this file :wink:

Reply:
Yes, I hoped you wouldn't notice, I hacked it in my local module, and used this method. If I fix one deprecation, I should do them all, which is work ;-)